comp/otelcol/otlp/components/statsprocessor/agent.go

```diff
@@ -36,14 +38,26 @@ type TraceAgent struct {
 	exit chan struct{}
 }
 
+type statsAdder interface {
```

Comment:
Can we add an implementation of the `Adder` interface here
```
type OTelStatsAdder struct {
	out chan *pb.StatsPayload
}

func (a *OTelStatsAdder) Add(payload *pb.StatsPayload) {
	a.out <- payload
}

func NewOTelStatsAdder(out chan *pb.StatsPayload) *OTelStatsAdder {
  return &OTelStatsAdder{out}
}
```

Then use this struct in `NewAgentWithConfig`. This way we don't need to introduce any breaking change at all.

Reply:
Yeah this works around this for now good idea here! At some point in the future once this is all released you should be able to migrate to a "channel-less" approach too